packages/api/cms-api/src/dependencies/dependencies.service.ts

```diff
@@ -64,17 +71,65 @@ export class DependenciesService {
+        );
         console.timeEnd("creating block dependency materialized view");
 
         console.time("creating block dependency materialized view index");
         await this.connection.execute(`CREATE INDEX block_index_dependencies_targetId ON block_index_dependencies ("targetId")`);
         console.timeEnd("creating block dependency materialized view index");
     }
 
-    async refreshViews(): Promise<void> {
-        console.time("refresh materialized block dependency");
-        await this.connection.execute("REFRESH MATERIALIZED VIEW block_index_dependencies");
-        console.timeEnd("refresh materialized block dependency");
+    async refreshViews(options?: { force?: boolean; consoleCommand?: boolean }): Promise<void> {
+        const refresh = async (options?: { concurrently: boolean }) => {
+            console.time("refresh materialized block dependency");
+            const blockIndexRefresh = this.refreshRepository.create({ startedAt: new Date() });
+            await this.refreshRepository.getEntityManager().persistAndFlush(blockIndexRefresh);
+
+            await this.connection.execute(`REFRESH MATERIALIZED VIEW ${options?.concurrently ? "CONCURRENTLY" : ""} block_index_dependencies`);
+
+            await this.refreshRepository.getEntityManager().persistAndFlush(Object.assign(blockIndexRefresh, { finishedAt: new Date() }));
+            console.timeEnd("refresh materialized block dependency");
+        };
+
+        if (options?.force) {
+            // force refresh -> refresh sync
+            await this.refreshRepository.qb().truncate();
+            await refresh();
+            return;
+        }
+
+        const lastRefreshes = await this.refreshRepository.find({}, { orderBy: { finishedAt: QueryOrder.DESC_NULLS_FIRST }, limit: 1 });
+        if (lastRefreshes.length === 0) {
+            // first refresh -> refresh sync");
+            await refresh();
+            return;
+        }
+
+        const lastRefresh = lastRefreshes[0];
+        // if the DB state indicates that a refresh takes more than 15 minutes,
+        // it's likely that the previous refresh isn't actually in progress but was interrupted unexpectedly
+        const isRefreshInProgress = lastRefresh.finishedAt === null && lastRefresh.startedAt > subMinutes(new Date(), 15);
+
+        if (isRefreshInProgress) {
+            // refresh in progress -> don't refresh
+            return;
+        } else if (lastRefresh.finishedAt && lastRefresh.finishedAt > subMinutes(new Date(), 5)) {
+            // newer than 5 minutes -> don't refresh
+            return;
+        } else if (lastRefresh.finishedAt && lastRefresh.finishedAt > subMinutes(new Date(), 15)) {
+            // newer than 15 minutes -> refresh async
+            const refreshPromise = refresh({ concurrently: true });
+            if (options?.consoleCommand) {
```

Comment:
name that `consoleCommand` not where it is called from, instead what it does, eg. `awaitRefresh` or `dontRunInBackground`